Docs(docfx): update docfx build locally guide and apply xref links

DevTKSS opened the pull request:
GitHub Issue (If applicable): closes #

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature
- Code style update (formatting)
- Documentation content changes
- Build or CI related changes (maybe?)
<!-- Please uncomment one or more that apply to this PR


- Refactoring (no functional changes, no api changes)

- Other... Please describe:
- Project automation


-->

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Current docfx guide is outdated and telling to use docfx.console which has been depreciated
see more Information to this here: https://github.com/dotnet/docfx/issues/9100
[Link to the Documentation page](https://platform.uno/docs/articles/uno-development/docfx.html)
The Uno.Extensions Repository is already having the new commands in its Readme.md

Secondary, the check_toc.ps1 does list almoast every href as not linked to a file, because it does not recognize the xref link declaration

Third page is way smaller changes:
- The [Getting-started](https://platform.uno/docs/articles/get-started.html) did not at all mention the need to install dotnet before Uno.Check (that is only told on the actual Uno.Check Page) so thats more a reason of completeness of the Quick Start.
- I did frequently notice Beginners ask about Trouble-Shooting Uno.Check and since we did already seperate its Pages (much better from my point of view!) there has been those links been missing.

## What is the new behavior?

<!-- Please describe the new behavior after your modifications. -->
1. Updated the docfx commands 

   - I did left there some commented text, which needs to be checked if the new one is enough.
     Problem with that is, I do not understand the UWPSyncGenerator so deeply to decide this. Currently, those comments do stretch the md line distance a bit (vs 2022 preview) althought its hidden.

2. Added a detailed guide to the usage of check_toc.ps1 along with adding the capability of recognizing and comparing the xref -> uids
  - By this it also did uncover the features/windows-xaml-ui-xbind link beeing broken, I did already wonder last days. Was unable to navigate to this page in the docs.
  - Second link it did uncover is the only url link in toc and to me it looks okay, but maybe someone would like to check this...
  - Another Link I do not know if the url is correct is the now nested Uno.Check Pages I tryed to link from the Getting-Started. 
    The old one does have a different signature...
  - Because of that new capability of the script, I added a -Verbose Flag for it, so in case of problems, there would be informations available, till which point it came.
  - **Help needed** with the Link to the gif I added to this for showing a workaround for those who might not know these steps until this point, to find not inluded files in vs 2022. I used the docfx documentation way for Image, but its showing only a broken image symbol...
  - I dont know if the script change does actually make this to a CI - Build relevant PR but there is told: ` At the moment it's not part of the CI,` so I dont know it should be tested for this again...

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results. -> building docfx is failing because of missing things it seems to me
[docfx build output.txt](https://github.com/user-attachments/files/19380577/docfx.build.output.txt)
- [x] Contains **NO** breaking changes -> would not identify it as such...
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). -> I did not see one just my problem with it.
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information

<!-- Please provide any additional information if necessary -->
Above mentioned

Internal Issue (If applicable):
<!-- Link to relevant internal issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

DevTKSS committed:
docs(get-started): Updated Quick Start
chore: Add Information about the v6 Extension improvements

DevTKSS committed:
docs(docfx): Update guide steps for building and serving the documentation locally

chore: Fixed outdated link for tabs to the docfx documentation

docs(tabs): Added information about ending tab content

docs(formatted-blockquotes): Added rendered samples for the mentioned formatted blockquotes beside note

chore: Fixed formatting and seperated contributors note to CI in note

DevTKSS committed:
docs(check_toc): Added guide to usage of check_toc script
chore: Listed both common ways to serve docfx output locally
docs(ImportExternalDocs): Added import of the README.md from Extensions Repository

This provides clarity of usage of the import_external_docs scripts available in the main documentation to lookup

unodevops commented:
⚠️⚠️ The build [158236](https://dev.azure.com/uno-platform///_build/results?buildId=158236) has failed on Uno.UI - docs.

DevTKSS commented:
The markdown linting pipeline in this case would be greatly more helpful if it would also lint and suggest the solution for image video and links like relative paths and so on before the intending and spaces checks👍🤔

unodevops commented:
⚠️⚠️ The build [158359](https://dev.azure.com/uno-platform///_build/results?buildId=158359) has failed on Uno.UI - docs.

DevTKSS commented:
Currently on Fixing a regex bug, then I will send the updated commits.
Had to delete the commented parts mentioned to satisfy the markdown lints, hopefully anyone could check then if there needs to be something added in relation to the UwpSyncGenerator.

DevTKSS commented:
Does a include of Pester tests required check the ui tests checkbook above?
Adding those for the functions of check_toc.ps1 which I now seperated (incl. Documentation of course) to check still listed toc items that I was quite sure that they are linked🤣
But since docfx is still failing to serve because of doc issues in the external nested stuff I am not able to review docs locally😒

agneszitte commented:
@DevTKSS  Thank you so much for the Pull Request ❤️
I will take the time to review the changes this week and follow up with you as needed

DevTKSS commented:
Status update:
Two of three seperated functions of the script to recognize xref and uid plus to compare them did pass successfully the tests

Things might to consider for add able tests and function for uid-xref links:
- casing
- minus between uid allowed but no spaces
- ...

Currently checking on the dot sourcing third script now which should be last before again check on the main script which uses them.

The dot sourcing one might be the most difficult one to setup test for because it has no actual function definition right now, just what it should execute. If it works but test setup has still trouble with its validation I would consider updating the related docs on it and commit to the remote branch so its review-able and maybe get a overseen solution for this.

Since I did add those md files, I did sort them into directory and recognized that not really matching and seperated directorys. Seems like there is the main contributing one and the uno-development one which mainly contains all of the files nested in toc contributing😅
@agneszitte is it considerable to move the documentation related ones at least with `git mv` so source control has no issues but this setup follows the toc definition?

DevTKSS committed:
docs(docfx): Updated doc files
chore(anchor-links): title moved out of metadata
chore(doc-templates): Added uid to templates for xref usage
chore(contributing): Updated most of the contribution specific md files directory to match their namespace

Co-authored-by: Agnès ZITTE <16295702+agneszitte@users.noreply.github.com>

DevTKSS commented:
@agneszitte commited to remote, unable to fix the Import-TocCheckerUtils.Test.ps1 to pass the check of Get-MarkdownHeader function, while it passed the dot-sourcing before. checking this here:
```
PS C:\Users\TKSSonja\Projects\uno\doc\articles\check-toc-utilities> . .\Get-MarkdownHeader.ps1
PS C:\Users\TKSSonja\Projects\uno\doc\articles\check-toc-utilities> Get-Command -Name "Get-MarkdownHeader" -CommandType Function

CommandType Name Version Source

Function Get-MarkdownHeader

PS C:\Users\TKSSonja\Projects\uno\doc\articles\check-toc-utilities>
```
It should have been able to get it from my point of view.
Could someone have a look on this and find the culprit?

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/index.html

DevTKSS commented:
Some planned to-do at the moment:
- [ ] 1. Enhance ./uno/doc/articles/index.md npm guide by adding tabbed content to choose from and include vs code extension options for this (seperate PR to be created, do we need a issue/kind-enhancement before?)
    - 1.1. cspell
    - 1.2. markdownlint
    - 1.3. adding links to ./uno/build dictionary/ preset files
- [ ] 2. After check-toc successfully passing:
    - 2.1. Check the now listed toc links
    - 2.2. Get Answer to the preferred link
       - [ ] xref ?
       - [ ] href with path?
    - 2.3. Add this to docs
    - 2.4. Match at least the now added pages to this 
    - 2.5. Add added doc files to the toc and link them in related topics if needed
- [ ] 3. Check the docfx issues with external docs in extension and workshops and samples

maybe toc checker can be adjusted to exclude the Readme files which are not cared about in samples, but preferred should be, to add explanation and how to each so it would greatly enhance others to learn from there.

The xref could be better especially because the external area do have quite long paths otherwise.

agneszitte commented on doc/articles/uno-development/docfx.md:
```diff
@@ -125,3 +144,7 @@ To run the script on Windows:
 3. Go to the uno\build folder (not the uno\src\build folder)
 4. Run the `run-doc-generation.cmd` script; make sure to follow the instructions
+
+## Import Uno Extensions and Tools docs
+
+[!INCLUDE ../external/  <!--https://github.com/unoplatform/uno.extensions/blob/main/doc/README.md -->]
```
to review and update

DevTKSS replied:
updated link, to be reviewed in Docs preview, my docfx is not willing to serve locally and writing way to much stacktrace, more than the terminal can handle (set to already 9000 lines!)

agneszitte commented on doc/articles/uno-development/docfx.md:
```diff
@@ -101,18 +120,18 @@ To run DocFX locally and check the resulting html:
-8. When DocFX runs successfully, it will create the html output at `C:\src\Uno.UI\docs-local-dist\_site`, which you can now view or mount on a local server.
+4. Open a Terminal at the Root Directory of your locally cloned Uno Repository.
+5. Install docfx globally: `dotnet tool install -g docfx`
+6. Run the following command: `docfx build doc/docfx.json` and attach any nested foldername you want by adding `-o your-nested-output-path`, default: `_site`
```
```suggestion
6. Run the following command: `docfx build doc/docfx.json` and attach any nested folder name you want by adding `-o your-nested-output-path`, default: `_site`
```

agneszitte commented on doc/articles/uno-development/docfx.md:
```diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
+# The Uno documentation website and DocFX
 
-Uno Platform's docs website uses [DocFX](https://dotnet.github.io/docfx/) to convert Markdown files in the [articles folder](https://github.com/unoplatform/uno/tree/master/doc/articles) into [html files](xref:Uno.Documentation.Intro).
+*[TOC]: Table of Contents
+*[HTML]: Hypertext Markup Language
+*[docs]: documentation
+*[doc file]: markdown file
```
to review this part

DevTKSS replied:
> to review this part

I did check it out at the staged docs and to me that behavior is quite suspicious... From their documentation thats definitely telling it would replace those parts, but somehow that does not apply to uno docs? Is that a markdown issue on their side or dies that only not apply in uno platform docs? In that or the other case, it seems to be not working as expected, so this docfx.json part should be decided to be cleared from there to avoid such things happening from my point of view...

DevTKSS replied:
Deleted the tags on commit #ce5524a8b29397ba69e61cd3345971ae29b451c4.htm

agneszitte commented on doc/articles/check-toc-utilities/Overview.md:
```diff
@@ -0,0 +1,59 @@
+---
+uid: Uno.Contributing.check-toc.Overview
+title: Overview
+---
+
+The `check_toc.ps1` script helps maintain the structure and integrity of your documentation by:
```
Missing proper title for the page here:
https://unodocsprstaging.z13.web.core.windows.net/pr-19722/articles/check-toc-utilities/Overview.html
![image](https://github.com/user-attachments/assets/5b80615a-4a8c-4751-bda9-3b1ad763b8cd)

agneszitte commented on doc/articles/check-toc-utilities/Guide.md:
```diff
@@ -0,0 +1,79 @@
+- If nothing is reported, verify the script was executed in the correct directory.
+- If `toc_additions.yml.tmp` is missing, check for file permission issues.
+
+   > [!NOTE]
+   > Visual Studio 2022 does not show the generated file by default.
+   > To open it, see these Steps:
+   > ![check-toc-find-toc-additions](..\uno-development\assets\check-toc-find-toc-additions-file.gif)
```
```suggestion
   > [!NOTE]
   > Visual Studio 2022 does not show the generated file by default.
   > To open it, see these Steps:
   >
   > ![check-toc-find-toc-additions](..\uno-development\assets\check-toc-find-toc-additions-file.gif)
```

https://unodocsprstaging.z13.web.core.windows.net/pr-19722/articles/check-toc-utilities/Guide.html#troubleshooting
![image](https://github.com/user-attachments/assets/f244c771-d928-4192-bb12-144b60a8d967)

agneszitte commented on doc/articles/toc.yml:
```diff
@@ -377,6 +379,7 @@
             href: uno-development/creating-runtime-tests.md
           - name: UI tests with Uno.UITest
             href: uno-development/creating-ui-tests.md
+          - name: Uno.Contributing.Scripts.PowerShell-Pester-Tests
```
```suggestion
          - name: Testing PowerShell Scripts with Pester
            href: Uno.Contributing.Scripts.PowerShell-Pester-Tests
```

DevTKSS commented on doc/articles/uno-development/doc-on-docs.md:
```diff
@@ -8,19 +8,19 @@ Good documentation is essential, and writing docs is a valued contribution that
-* how-to template: https://github.com/unoplatform/uno/blob/master/doc/.howto-template.md
-* WinUI feature template: https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md
-* Uno-only feature template: https://github.com/unoplatform/uno/blob/master/doc/.feature-template-uno-only.md
+* [how-to template](https://github.com/unoplatform/uno/blob/master/doc/.howto-template.md)
+* [WinUI feature template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md)
+* [Uno-only feature template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template-uno-only.md)
```
Can templates beginning with . Even be linked with xref? Thought not?

DevTKSS replied:
let's give it a try, added them in commit 05bf1c8

DevTKSS commented on doc/articles/uno-development/docfx.md:
```diff
@@ -73,26 +92,26 @@ Markdown:
-
-The [`check_toc` script](https://github.com/unoplatform/uno/blob/master/doc/articles/check_toc.ps1) checks for dead links in the TOC, as well as Markdown files in the 'articles' folder that are not part of the TOC. At the moment it's not part of the CI, but contributors can run it locally and fix any bad or missing links.
+> [!NOTE]
+> Use `---` in the Markdown sample is Important, to not include more Content in the tabbed area than actually wanted, but will not be rendered in the served documentation.
+> It is possible to use `***` alternatively for the same task.
```
In some doc files e.g. the templates, there is html tags used for seperation. Does that stand in relation to this formatting possibility? Maybe samples for this not direct markup format would be good to have included for completness?

DevTKSS replied:
applyed changes, to be checked at CI staged docs preview

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/index.html

DevTKSS commented:
work still in progress

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/index.html

DevTKSS commented on doc/articles/contributing/guidelines/creating-ps-pester-tests.md:
```diff
@@ -0,0 +1,26 @@
+uid: Uno.Contributing.Scripts.PowerShell-Pester-Tests
+---
+
+# Testing PowerShell Scripts with Pester
```
Switched namespace to `Tests` instead and applied to the other existing Uno.Contributing.Tests... simultaneously. So if that namespace would be better (would think it is) Scripts seperated might not explicitly needed.

DevTKSS commented:
@agneszitte three back- questions added to your comments, you might have an answer to? I am on doing throught file to file to validate and xref the links first in the contribution area so would be great if I could implement those changes then in one go:

First comment of yours that's having relevant question to this, would be the links to the dot prefixes templates you said to possibly update to xref.

I am not sure if linking that directly with xref would result in a valid link, does that work?

Second one is the abbreviation things which are somehow strangely do not show correctly as their docs are telling🤔

If that markdig extension does generally not work on uno docs, that should Maybe better get deleted from the docfx.json also beside not using it in the docs itself?

Third one, the possible scripts namespace

I would maybe better want to change to `Tests` together with the existing test docs also for brevity. The ps setup guide for ps script creation I could just add in the utils or add to the includes folder, because I could not really imagine a currently completely fitting one for this content. The option to "just" link to another website I do not prefer, because at the time I did research fir that information, there has been multiple videos telling they would show how to use ps7 in vs code, but none showed that really, so I would like to just fill that gaps up in that step-list there😅

DevTKSS commented on doc/articles/check-toc-utilities/Guide.md:
```diff
@@ -0,0 +1,79 @@
+
+---
+
+[!include[getting-help](includes/getting-help.md)]
```
added link as xref should now work, but unable to review locally

DevTKSS commented:
Where did now that uwp-conversions.md file suddenly come from, that's causing merge conflict...

unodevops commented:
⚠️⚠️ The build [159133](https://dev.azure.com/uno-platform///_build/results?buildId=159133) has failed on Uno.UI - docs.

unodevops commented:
⚠️⚠️ The build [159134](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159134) has failed on Uno.UI - CI.

DevTKSS commented:
toc.yml should not be checked for casing for xref usage

DevTKSS commented:
checked the "Run spell checking" pipeline problems, seems the CI needs some external help to ignore this or tell me and I switch the xref Names there, but since that are not my created ones and most likly be linked at some places, I would prefer decision from the team for this...
@agneszitte can you give me that advise you/uno prefers?

DevTKSS committed:
docs(docTemplates): Added uid to templates and contributing docs for xref usage

using xref makes the toc independant on where the file is actually placed

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
+```markdown
 > [!NOTE]
-> This is a Note, showing how to declare notes.
+> Information the user should notice even if skimming.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
```diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
-> This is a Note, showing how to declare notes.
-
-Use pre-formatted blockquotes when you want to call special attention to particular information.
+> Information the user should notice even if skimming.
```
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
```diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
 
 > [!IMPORTANT]
-> ...
+> Essential information required for user success.
```
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
-> ...
+```markdown
+> [!IMPORTANT]
+> Essential information required for user success.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
 
+```markdown
+> [!WARNING]
+> Dangerous certain consequences of an action.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
 ```
 
+> [!WARNING]
+> Dangerous certain consequences of an action.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
-> ...
+```markdown
+> [!TIP]
+> Optional information to help a user be more successful.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
```diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
 
 > [!TIP]
-> ...
+> Optional information to help a user be more successful.
```
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
````diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
+
+```markdown
+> [!CAUTION]
+> Negative potential consequences of an action.
````
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

DevTKSS commented on doc/articles/uno-development/docfx.md:
```diff
@@ -4,57 +4,76 @@ uid: Uno.Contributing.DocFx
 
 > [!CAUTION]
-> ...
+> Negative potential consequences of an action.
```
applied in [this commit](https://github.com/unoplatform/uno/pull/19722/commits/88110489ef82fa1b5f392ec128ec6f6bd1864758)

unodevops commented:
⚠️⚠️ The build [159337](https://dev.azure.com/uno-platform///_build/results?buildId=159337) has failed on Uno.UI - docs.

unodevops commented:
⚠️⚠️ The build [159338](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159338) has failed on Uno.UI - CI.

DevTKSS commented:
hoping there is nothing against seperations to fix that linted xref parts

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19722/index.html

DevTKSS commented:
failing because the toc i already linked is included in this mentioned PR for Uno.Samples:
https://github.com/unoplatform/uno/pull/19722#issuecomment-2767547912

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19722/index.html

unodevops commented:
⚠️⚠️ The build [159509](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159509) has failed on Uno.UI - CI.

unodevops commented:
⚠️⚠️ The build [159497](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159497) has failed on Uno.UI - CI.

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19722/index.html

unodevops commented:
⚠️⚠️ The build [159940](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159940) has failed on Uno.UI - CI.

unodevops commented:
🤖 Your WebAssembly Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-19722/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/index.html

unodevops commented:
⚠️⚠️ The build [159995](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=159995) has failed on Uno.UI - CI.

unodevops commented:
⚠️⚠️ The build [168896](https://dev.azure.com/uno-platform///_build/results?buildId=168896) has failed on Uno.UI - docs.

unodevops commented:
⚠️⚠️ The build [168908](https://dev.azure.com/uno-platform///_build/results?buildId=168908) has failed on Uno.UI - docs.

DevTKSS commented:
@booskander splitted toc checker from this, dont know if there is any interest in the left contents as I did just update kind of all href file links to xref so the actual folder structure would not matter any longer for the links to work... Maybe I should just delete all branches as my PR are just getting forgotten if someone of the team is doing same.

DevTKSS committed:
docs(templates): Remove very outdated information

Rider Android Support is reported as incompatible as of 06.02.2025 should get checked and until then removed

DevTKSS committed:
chore(PasswordVault): Adjust pipe table markdown and trailing whitespaces

DevTKSS committed:
chore: Check broken links and information to be current in Readme

DevTKSS committed:
chore(testingPS): update Links and remove dropped commits content links

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html

DevTKSS committed:
chore: update doc Readme for contributor markdown and information about vs code extensions

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html

DevTKSS commented:
@agneszitte @Jen-Uno @MartinZikmund (as you reviewed initially)
finalized this PR and its smallered its commit count and containment. From my side Ready to be reviewed 👍 

## Check this comments for a overview:

- [TOC of this PR](https://github.com/unoplatform/uno/pull/19722#issuecomment-2753600641)
- [uid Namespace Renamings](https://github.com/unoplatform/uno/pull/19722#issuecomment-2770090759)

## To be checked from the Team:

**These Pages/Files are obviously including outdated information, as of v6 Major Version:**
- [Passwort Vault doc page](https://github.com/unoplatform/uno/blob/master/doc/articles/features/PasswordVault.md) - Platform Win 7 and targets. [windows docs](https://learn.microsoft.com/de-de/uwp/api/windows.security.credentials.passwordvault?view=winrt-26100) are saying its implemented for windows (just like our docs), but that incapability is filed for Skia Desktop so thats something I can not difference to be possible with our new Win32
- [unoplatform/uno Github Main Readme](https://github.com/unoplatform/uno?tab=readme-ov-file#uno-platform-features) seems like miss information as by the 6.0
    I added a Comment to the appropriate section, so you can check. As this is the uno platform advertising point No 1 beneath the Homepage, it would make defintly sense, to add the great news for Hot Design and SkiaRenderer also there and we have the win 7 there also still filed as available platform
- You do file in the Readme
    > The [SQLite + Entity Framework Core App](https://sqliteefcore-wasm.platform.uno/), a demo of the combination of [Roslyn](https://github.com/dotnet/roslyn), [Entity Framework Core](https://learn.microsoft.com/ef/core/), [SQLite](https://github.com/unoplatform/uno.SQLitePCLRaw.Wasm) and the Uno Platform to manipulate an in-browser database.
    But this is not having current documentation for a how to beside the Migration from Silverlight docs, that the search index of the docs would show? I would indeed LOVE to have an option or/and guide for using a (maybe also MySql or similar DB), because I am searching for this right now, to integrate it on my serverside-wasm target ❤️

DevTKSS commented on doc/articles/contributing/guidelines/updating-dependencies.md:
```diff
@@ -22,7 +22,7 @@ The following dependencies don't change the public API surface and are typically
 
 The following dependencies require manual adjustments before merging:
-
+<!-- TODO: decide if following line should be changed, since docfx.console is officially depreciated! -->
 - [docfx.console](https://github.com/unoplatform/Uno/pull/1082/commits/c222caf8c23b35e19f6b33cd624cbfa714250bfe)
```
@booskander could you check on this?

DevTKSS commented on doc/README.md:
```diff
@@ -45,7 +45,7 @@ npm install
 ## Generated Implemented views
 
-The process of generating implemented views is documented on this page. [Building docs website locally with DocFX](https://platform.uno/docs/articles/uno-development/docfx.html?tabs=tabid-1#building-docs-website-locally-with-docfx).
+The process of generating implemented views is documented on this page. [Building docs website locally with DocFX](xref:Uno.Contributing.DocFx#building-docs-website-locally-with-docfx).
```
@booskander then the naming here for the xref is also to be fitted?

DevTKSS replied:
applyed casing fix also for uid in https://github.com/unoplatform/uno/pull/19722/commits/003610c32fe5d1c2a7c542c3e84ebeb29f17e100

DevTKSS committed:
chore: Update link

DevTKSS committed:
chore(DocFx): Apply casing update to match lowercasing
chore: moved the gif file to the assets folder near to the file its used with
chore: added the Anchor Links Link to the toc

used replace in vs code which should have found all occurrences

DevTKSS commented on doc/.howto-template.md:
```diff
@@ -32,21 +36,17 @@
-  * [.NET Core 3.1 SDK](https://dotnet.microsoft.com/download/dotnet-core/3.1) (**version 3.1.8 (SDK 3.1.402)** or later)
-  * [.NET Core 5.0 SDK](https://dotnet.microsoft.com/download/dotnet-core/5.0) (**version 5.0 (SDK 5.0.100)** or later)
+* **.NET SDK**
+  * [latest .NET SDK](https://dotnet.microsoft.com/download/latest)
```
checked in https://github.com/unoplatform/uno/pull/19722/commits/1ef10a8361f27974b751c7f1ff4997e53bdde9c4

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html

DevTKSS commented:
@jeromelaban would guess here is just that one comment to be checked and seems like bootscander is out of office, do you see anything left to do here?

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html

DevTKSS committed:
chore: Move getting started per IDE and extension to the getting started folder to fix the image links

Also updated the `/includes/` links to keep the file sourcing working

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-19722/docs/index.html
